# Introduction to Comparison Operations

Comparison operators allow us to compare values on either side of the operator for numeric values. If we have values that are of mixed types, one of the values will the coerced up the nmeric tower from an integer to a float, or a float to a complex. The result of the comparison is a Boolean (True or False).

[There are 8 comparison operations in Python](https://docs.python.org/3/library/stdtypes.html#comparisons) but we will focus on the 6 operators that are used frequently for analytics:

| Operator | Description | Example | Result |
| :--- | :----------- |:--- | :----------- |
|<|less than|`10 < 20`|`True`
|<=|less than or equal to|`10 <= 20`|`True`
|>|greater than|`10 > 20`|`False`
|>=|greater than or equal to|`10 >= 20`|`False`
|==|equal|`10 == 20`|`False`
|!=|not equal|`10 != 20`|`True`



Comparison operators can be used in a simliar fashion when comparing things like variables or strings as you'll see below. 

## Numeric comparison


In [38]:
134>34

True

## Variable comparison

In [23]:
a = 10
b = 75
c = 100

if (a == b):
    print('a is equal to b')
else:
    print('a is not equal to b')
    
print('----------------------------')
    
if (a < c):
    print('a is less than to c')
else:
    print('a is not less than to c')
    
print('----------------------------')

if (b > c):
    print('b is greater than to c')
else:
    print('b is not greater than to c')

a is not equal to b
----------------------------
a is less than to c
----------------------------
b is not greater than to c


## String comparison

When using >, >=, <, or <=, Python is comparing the length or number of characters between both strings


In [30]:
'Adrian'>='Javier'

False

In [33]:
'Adrian'<='Javier'

True

When using == or !=, Python compares the the two strings character by character

In [35]:
'Adrian'=='Adrian'

True

In [36]:
'adrian'=='Adrian'

False

In [37]:
'adrian'!='Adrian'

True

# Introduction to Arithmetic Operations

You can perform arithmetic operations like addition, subtraction, multiplication, and division.

The following are arithmetic operators that Python supports:

| Operator | Description | Syntax | Example |Result|
| :--- | :----------- |:--- | :----------- |:----------- |
| + | Addition | x+y |1 + 2|3
| - | Subtraction |x-y|2 - 1|1
| * | Multiplication |x*y|2 * 2|4
| / | Division (float) |x/y|100 / 9|11.11..
| // | Division (floor) |x // y|100 // 9|11
| % | Modulus - Returns the remainder when left operand is divided by the right operand |x % y|100 % 9|1
| ** | Power |x**y|2 ** 5|32

In [14]:
# Addition
52+10

62

In [40]:
# Subtraction
52-10

42

In [42]:
# Multiplication
52*10

520

In [43]:
# Division
52/10

5.2

In [45]:
# Modulus
# Commonly used to tell if a number is even or odd
52%10

2

In [46]:
# Power
52**10

144555105949057024

## Solving a problem using variables

Let's try to solve the following problem using variables
> A clothing store sells t-shirts for $10 and makes 25% profit from each sale. If the store sells 500 t-shirts this weekend, how much total profit does it make?

Seems like a relatively straightforward calucation

In [47]:
500 * (.25 * 10)

1250.0

So the clothing storee will make a total profit of $1250. While this is a straightfoward way to solve this problem, looking at the code, it's not clear what each number represents. One way of solving this problem is by assigning each value into a variable.

In [48]:
cost_of_shirt = 10
profit_margin = .25
number_of_shirts_sold = 500

After running the cell above, our three variables now conatin the information provided in the word problem. We check that our variables contain the correct values by running each of them

In [49]:
cost_of_shirt

10

In [50]:
profit_margin

0.25

In [51]:
number_of_shirts_sold

500

Let's combine our variables using the arithmetic operations to create a new variable we will call `total_profit`

In [52]:
total_profit = number_of_shirts_sold * (profit_margin * cost_of_shirt)

In [53]:
total_profit

1250.0

Let's use the `print` function to display a clear message of what our stores total profit is

In [54]:
print("The clothing store made a total profit of $", total_profit)

The clothing store made a total profit of $ 1250.0
